From e325ac12b87f28346e5c04b09fa513665f2ea048 Mon Sep 17 00:00:00 2001
From: Robert Mustacchi <rm@joyent.com>
Date: Thu, 8 Dec 2016 23:46:18 +0000
Subject: [PATCH] OS-5840 ixgbe crashes on debug builds when it fails to attach
 Reviewed by: Patrick Mooney <patrick.mooney@joyent.com> Reviewed by: Ryan
 Zezeski <ryan.zeseski@joyent.com> Reviewed by: Joshua M. Clulow
 <jmc@joyent.com> Approved by: Patrick Mooney <patrick.mooney@joyent.com>

---
 usr/src/uts/common/io/ixgbe/ixgbe_main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/common/io/ixgbe/ixgbe_main.c b/usr/src/uts/common/io/ixgbe/ixgbe_main.c
index c77bfece05..1bf0a8d520 100644
--- a/usr/src/uts/common/io/ixgbe/ixgbe_main.c
+++ b/usr/src/uts/common/io/ixgbe/ixgbe_main.c
@@ -846,7 +846,9 @@ ixgbe_unconfigure(dev_info_t *devinfo, ixgbe_t *ixgbe)
 	/*
 	 * Unregister interrupt callback handler
 	 */
-	(void) ddi_cb_unregister(ixgbe->cb_hdl);
+	if (ixgbe->cb_hdl != NULL) {
+		(void) ddi_cb_unregister(ixgbe->cb_hdl);
+	}
 
 	/*
 	 * Remove driver properties
-- 
2.21.0

